### PR TITLE
DAOS-3245 control: doc updates and unit tests for storage prepare

### PR DESCRIPTION
**Pull request by tanabarr**
Follow up changes after introducing new storage prepare syntax.

* update admin guide and README with correct syntax
* improve unit test coverage
* provide more accurate cli warnings for relevant operations
* remove pre-existing memory allocations before setting
* improve prep rpc test coverage, add PrepScm.GetState()

### CODE REVIEW

File: src/control/client/connect.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (cr ClientResult) String() string {
 	if cr.Err != nil {
 		return fmt.Sprintf("error: " + cr.Err.Error())
 	}
-	return fmt.Sprintf("%+v", cr.Value)
+	return fmt.Sprintf("%#v", cr.Value)
```
Comment (mjmac) [inline src/control/client/connect.go]: Is this string only intended to be used for debugging?

File: src/control/cmd/daos_server/storage.go [hunk 2]
```diff
@@ -107,13 +107,24 @@ func (cmd *storagePrepareCmd) Execute(args []string) error {
 		}
 	}
 
-	if cmd.ScmOnly || !cmd.NvmeOnly {
+	for cmd.ScmOnly || !cmd.NvmeOnly {
```
Comment (mjmac) [inline src/control/cmd/daos_server/storage.go]: This is a confusing construction and requires extra thought as to what is happening here. In fact, I've been looking at it for a couple of minutes and I still don't quite understand why it's a loop now.

Suggest refactoring these blocks to be controlled by new booleans, e.g.:
```go
prepScm := cmd.ScmOnly || !cmd.NvmeOnly
prepNvme := cmd.NvmeOnly || !cmd.ScmOnly
```

The benefit here is that there is clarity about when a block will be entered, and it allows you to potentially do other more complex logic to determine when those booleans should be set to true (e.g. probing storage state, etc).

**Comment by mjmac** [comment]
I think it would be best to pull the doc updates out into a separate PR and land them ASAP. The rest of the work is going to need a bit more time.

**Comment by mjmac** [comment]
Also, if you're going to move the SCM stuff around, might as well go all the way and move it under server/storage, as that's the long-term goal anyhow. We want to isolate the code which needs privileged access, ideally in packages other than server/ so that our privileged binary does not need to import anything in the server package.

File: src/control/cmd/daos_server/storage.go (hunk 2 above)
Comment (tanabarr) [reply src/control/cmd/daos_server/storage.go]: it's a loop so can be broken out of

File: src/control/client/connect.go (hunk 1 above)
Comment (tanabarr) [reply src/control/client/connect.go]: placeholder for more useful output of generic objects before I get round to finding a proper solution

**Comment by tanabarr** [comment]
split into 3 separate PRs as requested:
https://github.com/daos-stack/daos/pull/1073
https://github.com/daos-stack/daos/pull/1074
https://github.com/daos-stack/daos/pull/1075

**Pull request closed without merging**